src/userSurvey.js

```diff
@@ -185,7 +209,7 @@ export async function setSurvey(options) {
     }
 
     // If this is the first survey request, open the survey in a new tab
-    if (lastSurveyRequest === null) {
+    if(lastSurveyRequest === null) {
```

Comment:
We don't want to generate a new survey ID for each survey. We want one survey ID for the duration of the study.

Reply:
I moved the functionality of generating a new survey ID into the `getSurveyId` method. This method now generates a survey ID if there is no value for survey ID in storage.